### DISCUSSION TITLE
An issue with the GPU modes on Zephyrus G15 2022

### DISCUSSION THREAD

**Discussion by ComradeMP** [seerge/g-helper#894] [General]
I'd like to begin by saying G-Helper is one of the best utility apps i've ever seen, it's impressive how much development it had in such short time frame and the amount of functions it provides, it's an insult to compare it to Asus's shitty software.

Anyway, I'm experiencing an issue with this software, particularly related to Eco mode, I didn't have this problem with the Armoury Crate.

I own a Zephyrus G15 2022 (GA503RW). If I shut down my laptop with Eco or Optimized mode enabled and turn the laptop on again later, the dGPU will always be enabled and cannot be disabled unless I restart my laptop, then the GPU modes start to function properly again. I never had this issue with Armoury Crate, I could just turn the laptop on and it would turn on with dGPU disabled if it was shutdown on Eco mode (or unplugged/optimized).

Here's another note:
So for some reason, on my model (not sure if it's just mine), when the dGPU is turned on regardless of whether ultimate mode is enabled or not, it appears as a detachable media, and I can eject it like a flash drive. When I switch to eco mode and turn on my laptop on battery, the GPU appears under the name "Video Controller (VGA Compatible)", and there seems to be no way to turn it off without restarting the laptop. When I restart the laptop, this issue goes away, but reppears later under the conditions I mentioned above. Normally it appears as "NVIDIA GeForce RTX 3070Ti Laptop GPU" in the detachable media dialogue.

None of these issues existed when I used Armoury Crate.
Is it possible to address this issue for this model?

**Comment by seerge** [comment]
@ComradeMP hello. This is literally a duplicate of this issue https://github.com/seerge/g-helper/issues/878 (and couple of others)

G15 2022 has notoriously bugged bios. And people report issues with Eco mode for both AC / G-helper (as this is not app related). If you believe AC works for you - it's absolutely fine to use it :) 

As for "detachable media", this is how windows reacts on newly appearing device in device manager (this happens when you enable dGPU). Clicking on eject is harmless, and won't do anything. For me it happens if you enable GPU in AC as well.

**Reply by ComradeMP** [reply]
Oh hell no I'm not going back to AC, it would be a downgrade haha. The benefits of G-Helper are clear as day (lower temps, faster logins, easy to use UI, etc.).

Is there any way to solve the bios problems other than waiting for Asus to make it worse?

**Reply by seerge** [reply]
@ComradeMP you can try to revert to 311 (or what is previous there for your model). It's available on Asus Support page. Someone reported it worked for them.

**Reply by ComradeMP** [reply]
I reverted to 311 just now. I can't replicate the issue after the bios downgrade.
Thank you for the quick responses by the way. You're the best 👍

**Reply by seerge** [reply]
Great, I will keep this discussion open as a reminder for other G15 users :)

**Reply by Shubham-Singh-01** [reply]
This issue is also present in Asus Tuf A15 2021 model, everything seems to works before shutdown, and if you try to change modes after that it will through a BSOD error (VIDEO DXGKRNL FATAL ERROR), Asus didn't fixed this issue and also I didn't find any solution for this.

**Reply by seerge** [reply]
@Shubham-Singh-01 try to check 'Enable GPU on shutdown' https://github.com/seerge/g-helper/discussions/1042 it may help

**Reply by Shubham-Singh-01** [reply]
Thanks:)
